tests/database/helpers/EventAccumulator.ts

```diff
@@ -0,0 +1,30 @@
+export class EventAccumulator {
+  public eventData = [];
+  public promise;
+  public resolve;
+  public reject;
+  constructor(private expectedEvents: number) {
+    if (!this.expectedEvents) throw new Error('EventAccumulator:You must pass a number of expected events to the constructor');
+    this.promise = new Promise((resolve, reject) => {
+      this.resolve = resolve;
+      this.reject = reject;
+    });
+  }
+  addEvent(eventData?: any) {
```

Comment:
FWIW, "any" is considered dangerous (see the google3 style guide on it here: https://g3doc.corp.google.com/javascript/typescript/g3doc/styleguide.md?cl=head#any-type) and should generally be avoided.  You could use a more specific type ("object" or "object|number|string" or something), but even better might be to make EventAccumulator generic (i.e. make it "EventAccumulator<T>", this method would take a "eventData: T" and this.promise would be a Promise<T[]>, etc.  Then the events are typed all the way through.  That's definitely how I'd write it from scratch, but if it complicates the migration, it's okay to do something easier.